### PR TITLE
Version 21.0 Remove coverletter,categories,employmnettypes,workstatuses and add datalist client, browserid client and refactor test classes

### PR DESCRIPTION
**Pull request by usbsnowcrash**
## 21.0
This version represents a major overhaul both in how we call our APIs as well as which versions of the APIs we are going to use. 

### Previous PRs:
https://github.com/careerbuilder/ruby-cb-api/pull/240 Removing Coverletters and old resume list
https://github.com/careerbuilder/ruby-cb-api/pull/241 Test refactoring
https://github.com/careerbuilder/ruby-cb-api/pull/243 Adds new exception types
https://github.com/careerbuilder/ruby-cb-api/pull/244 Adds browser id call
https://github.com/careerbuilder/ruby-cb-api/pull/246 Adds datalist APIs and removes old ones

### CODE REVIEW

**Comment by ausmith** [comment]
@usbsnowcrash I'm going to re-open this PR as I believe all the dependent PRs are merged into this branch. Feel free to close if you feel otherwise.

**Comment by oochelz** [comment]
I think this one needs opened next: https://github.com/careerbuilder/ruby-cb-api/pull/245

**Comment by oochelz** [comment]
I'm going to let Jeff sort it out and reopen if he feels it's time. I'm iffy on that one PR, since it was closed (not merged).

**Comment by ausmith** [comment]
@oochelz agreed that #245 is wonky as it indicates closed at the top but merged if you scroll to the bottom?

File: lib/cb/clients/data_list.rb [hunk 1]
```diff
@@ -26,6 +26,14 @@ def query_strings(args)
...
           qs
         end
+
+        def headers(args)
+          {
+              'Accept' => 'application/json;version=2.0',
+              'Authorization' => "Bearer #{ args[:oauth_token] }",
+              'Content-Type' => 'application/json'
+          }
+        end
```
Comment (oochelz) [inline lib/cb/clients/data_list.rb]: I thought this came with `Base`?

Comment (usbsnowcrash) [inline lib/cb/clients/data_list.rb]: This one is different than base because its asking for version=2.0

File: lib/cb/version.rb [hunk 2]
```diff
@@ -9,5 +9,5 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and limitations under the License.
 module Cb
-  VERSION = '20.6.0'
+  VERSION = '21.0.0.pre1'
```
Comment (oochelz) [inline lib/cb/version.rb]: PRE!

Comment (usbsnowcrash) [inline lib/cb/version.rb]: whoops

**Pull request merged**